build.sh

```diff
@@ -0,0 +1,270 @@
+        ;;
+    "-d" | "--debug")
+        debug=1
+        ;;
+    "-g" | "--gdb")
+        gdb=1
+        ;;
+    "-a" | "--analyze")
+        analysis=1
+        ;;
+    "-v" | "--variant")
+        shift
+        variant="$1"
+        ;;
+    *)
+        error "unknown parameter: $1"
+        ;;
+    esac
+    shift
+
+done
+
+# not supported combinations of flags?
+(( $speculos )) && (( $load )) && error "-l and -s cannot be used at the same time"
+
+# map device to model (different in speculos)
+case "$device" in
+    nanos )
+        model="nanos"
+        ;;
+    nanox )
+        model="nanox"
+        ;;
+    nanosplus )
+        model="nanosp"
+        ;;
+    *)
+        error "unknown device: $device"
+        ;;
+esac
+
+# find SDK version number
+BOLOS_SDK="$device-secure-sdk"
+
+[ ! -f "./dev/sdk/$device-secure-sdk/Makefile.defines" ] && error "sdk not found. Are the submodules initialized?"
+
+# get sdk version from sdk
+sdk="$( grep '^#define BOLOS_VERSION' ./dev/sdk/${device}-secure-sdk/include/bolos_version.h | awk '{ print $ 3}' | tr -d '"' )"
+
+# if the first character is a digit, we are convinced it's a valid version number
+grep -q '^[[:digit:]]' <<< "$sdk" || error "$sdk not a valid version"
+
+# find the fitting cxlib of speculos
+[ -z "$cxlib" ] && {
+    # we assume we have the same cxlib as the sdk
+    cxlib="$sdk"
+
+    while [ ! -z "$cxlib" ]
+    do
+        cxlib_fn="./dev/speculos/speculos/cxlib/$model-cx-$cxlib.elf"
+        [ -f "$cxlib_fn" ] && {
+            # we found something matching
+            break
+        }
+        # we iteratively remove the last component of the version
+        # 1.0.2 -> 1.0
+        # 1.0 -> 1
+        # 1 -> ""
+        cxlib="$( awk -F'.' 'BEGIN{OFS="."} NF{NF--};1' <<< "$cxlib" )"
+    done
+}
+
+# if it is zero, we didn't find something matching
+[ -z "$cxlib" ] && error "no fitting cxlib found. Try -c|--cxlib."
+
+# yay, finally we can start
+echo "device $device selected, sdk $sdk found, using cx-lib $cxlib"
+
+# build the app
+# pull and tag image
+pull_image \
+    "$IMAGE_BUILD" \
+    ledger-app-builder || error "couldn't pull image"
+
+build_flags=""
+
+# if speculos requested, add the flag
+(( $speculos )) && {
+    build_flags+="SPECULOS=1 "
+}
+
+# if debug requested, add the flag
+(( $debug )) && {
+    build_flags+="DEBUG=1 "
+}
+
+[ ! -z "$variant" ] && {
+    build_flags+="CHAIN=$variant "
+}
+
+extra_args=""
+
+(( analysis )) && {
+    build_flags+=" scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs "
+    extra_args+="-v /tmp:/app/scan-build "
+}
+
+# default make cmd
+cmd="make clean && $build_flags make "
+
+# we have to map usb into the docker when loading the app
+(( $load )) && {
+    extra_args+="--privileged -v /dev/bus/usb:/dev/bus/usb "
+    cmd+="&& $build_flags make load"
+}
+
+docker run \
+    -e BOLOS_SDK="/app/dev/sdk/$BOLOS_SDK" $extra_args \
```

Comment:
Hmm, because I thought different devices would need different SDKs and I don't know how different SDKs are handled inside the docker image :thinking: I can try to figure out if it is really needed :thinking:

Reply:
Oh, it works without env variable :thinking: :man_shrugging: Will be removed :grimacing:

Reply:
The reason is here explained: https://github.com/LedgerHQ/app-iota/pull/6#discussion_r1086220366

Reply:
changed in: https://github.com/iotaledger/ledger-iota-app/pull/34